microsoft/testsuites/mdatp/mdatp.py

```diff
@@ -22,6 +23,56 @@
 EXIT_ONBOARD_INFO_FOUND = 255
 
 
+def ensure_working_path_allows_executables(node: Node) -> None:
+    # EXAMPLE WORKAROUND:
+    # - Get the mount point for the path.
+    # - Check if the mount point is mounted noexec.
+    # - if yes, remount with exec flag
+    working_path = node.get_working_path()
+    mount_info = node.tools[Df].get_partition_by_path(directory=working_path.as_posix())
+    if not mount_info:
+        # info only, since we have not validated whether the working path is executable.
+        # An image with weird DF or Mount output could still be able to pass, so we
+        # will info log and allow the test to continue.
+        node.log.info(
+            f"Could not locate partition info for directory "
+            f"{working_path.as_posix()}. "
+            "Test may fail due to noexec permissions error, "
+            "or due to the working path not existing."
+        )
+        return
+    mountpoint = mount_info.mountpoint
+    partitions = node.tools[Mount].get_partition_info(mountpoint=mountpoint)
+    if not partitions:
+        # info only, since we have not validated whether the working path is executable.
+        # An image with weird DF or Mount output could still be able to pass, so we
+        # will info log and allow the test to continue.
+        node.log.info(
+            f"Could not locate mount info for directory "
+            f"{working_path.as_posix()}. "
+            "Test may fail due to noexec permissions error, "
+            "or due to the working path not existing."
+        )
+        return
+    # else, we will check the permissions and fix as needed
+    partition = partitions[0]
+    if "noexec" in partition.options:
+        node.log.info(f"Working path in {mount_info.mountpoint} is mounted noexec!!")
+        # handle bsd/linux differences in mount command
+        if isinstance(node.os, BSD):
+            # fetch all mount options for BSD and omit 'noexec'
+            options = [option for option in partition.options if option != "noexec"]
+        else:
+            # Linux allows remounting as exec with other options preserved.
+            # just have to pass 'exec' to remount
+            options = ["exec"]
+        # initiate the remount.
+        node.tools[Mount].remount(point=mountpoint, options=options)
+
+    # otherwise, no issues with partition, nothing to fix.
+    return
```

Comment:
It doesn't need to return at the end, if no return parameter defined.

Reply:
Isn't this more of a style choice then? Removed.